frontend/packages/metal3-plugin/src/components/host-detail.tsx

```diff
@@ -178,19 +190,33 @@ const BaremetalHostDetails: React.FC<BaremetalHostDetailsProps> = ({
   );
 };
 
-export const BaremetalHostDetailPage: React.FC<BaremetalHostDetailPageProps> = (props) => {
-  const machinesResource = {
-    kind: referenceForModel(MachineModel),
-    namespaced: true,
-    isList: true,
-    prop: 'machines',
-  };
-  const nodesResource = {
-    kind: NodeModel.kind,
-    namespaced: false,
-    isList: true,
-    prop: 'nodes',
-  };
+const BaremetalHostDetailPage: React.FC<BaremetalHostDetailPageProps> = (props) => {
+  const { hasNodeMaintenanceCapability } = props;
+  const resources: FirehoseResource[] = [
+    {
+      kind: referenceForModel(MachineModel),
+      namespaced: true,
+      isList: true,
+      prop: 'machines',
+    },
+    {
+      kind: referenceForModel(NodeModel),
+      namespaced: false,
+      isList: true,
+      prop: 'nodes',
+    },
+  ];
+
+  if (hasNodeMaintenanceCapability) {
```

Comment:
FYI, this can also be solved via dynamic array declaration pattern:

```ts
const resources: FirehoseResource[] = [
  {
    /* stuff */
  },
  ...(hasNodeMaintenanceCapability && [{
    /* stuff */
  }] || []),
];
```

But I'm also OK with proposed code.